### ISSUE TITLE
Can't seem to enable Cardboard XR Plugin in Unity

### ISSUE THREAD

**Issue by JJNCreator** [googlevr/cardboard#115]
I installed this plugin today, and it looks like Unity is having trouble enabling it. I click on the checkmark next to Cardboard XR Plugin in XR Plugin Management, then it says "Assigning all requested folders", then stops with the check remaining unchecked. When I play the game, it comes back with this warning:

**Cardboard XR Plugin package Initialization not completed. You will need to create any instances of the loaders and settings manually before you can use the intended XR Plug-in Package.**

Is this a bug, or am I doing something wrong? Thanks

By the way, I'm running Unity 2019.4.1f1 on macOS

**Comment by jballoffet** [comment]
Thanks for reporting the issue. I suspect it could be related to a misconfiguration in Unity project settings (e.g. not enabling Cardboard XR Plugin in XR Plug-in Management settings). I’d suggest that you carefully follow the steps described in the [quickstart webpage](https://developers.google.com/cardboard/develop/unity/quickstart). If you’re still unable to run the sample app provided with it, please reach out to us again, providing a sample project where the issue is shown and the steps required to reproduce it.

**Comment by jballoffet** [comment]
I'm closing this issue, please reopen if it persists.

**Comment by craigcomstock** [comment]
I had to find the XR Plug-in Management settings, under Edit -> Project Settings, XR Plug-in Management settings, then click on the Android icon tab and check the checkbox for Cardboard XR Plugin. Not sure if things work after that, but am checking.

Installing with the .git URL failed so I just cloned the https://github.com/googlevr/cardboard-xr-plugin repo and imported from disk the package.json file, then it said in the Window -> Windows -> Console

```
Cardboard XR Plugin package Initialization not completed. You will need to create any instances of the loaders and settings manually before you can use the intended XR Plug-in Package.                           
UnityEditor.EditorApplication:Internal_CallUpdateFunctions() (at /home/bokken/buildslave/unity/build/Editor/Mono/EditorApplication.cs:310)
```

**Comment by craigcomstock** [comment]
@JJNCreator @jballoffet I tried this same thing with linux appimage on debian 10 with unity 2019.4.16f1. I tried adding direct from git cloned repo and it fails as originally supported. I then tried checking the checkbox and re-try the import, still fails.

I thought maybe I needed to enable the XR Plug-in Management for Cardboard FIRST before trying to add the repo.

I went Edit -> Project Settings, XR Plugin Management and clicked on the "Install XR Management" button. This finished with no errors. Then I looked for the checkbox for Cardboard and there was none to check.

I then imported from disk the Cardboard XR Plugin and ... it still failed. :+1: 

Can you re-open the ticket or comment how you worked around the issue?

**Comment by craigcomstock** [comment]
I tried refreshing the packages in the Package Manager and closing and re-opening the Unity IDE. Neither seem to work to get the Google Cardboard XR Plugin available for the next step in the quickstart: https://developers.google.com/cardboard/develop/unity/quickstart

I also tried downloading the releases https://github.com/googlevr/cardboard-xr-plugin/releases, those .tar.gz files don't work with the Import from Tarball option in Unity, but unpacking the tar.gz file for 1.4.0 and importing from disk had no complaints, but still did not present the Cardboard XR Plugin in the list of packages.

**Comment by Treecrotch** [comment]
@craigcomstock 

Hi. did you try the source code **.zip** from that last link you posted?
https://github.com/googlevr/cardboard-xr-plugin/releases

I'm using unity 2020.1.13f1 on windows 10, with this setup I can confirm _"add package from disk"_ was successful for me to build the project and deploy to android

**Comment by craigcomstock** [comment]
Oh! In the top-left of Package Manager the second drop-down has 'Unity Registry' by default. When I select 'In Project' I can see Google Cardboard XR Plugin for Unity!

Import into project button pressed, and no errors in Window -> Windows -> Console!

In the instructions:

```
Navigate to Assets/Samples/Google Cardboard/<version>/Hello Cardboard/Assets/Scenes, select Add Open Scenes, and choose HelloCardboard to open the sample scene.
```

I assume that is navigate in the Project tab (Window -> Windows -> Project).

I didn't see anything that said Add Open Scenes so I just double-clicked on the HelloCardboard scene.

It seems things are OK from there so far.

So I suspect the trick is DON'T follow the quick start instructions about the git URL but rather download a release. I could do a check on the diff between 1.4.1 and master to see what changed. Maybe even just git checkout 1.4.1 and see if that works.

Thanks @Treecrotch !

**Comment by Treecrotch** [comment]
Well all I can tell you is I ran in to similar issues at first, but now I can get it working w a new project pulling it in from git as it suggests, or doing it w the zip file locally as you just did. So both ways do work for me now.

As far as the "hello cardboard" part you quoted from the quick start.

I think what it means is that, yes, in the project tab search for and open the hello cardboard sample scene. 
but also in build settings, you must have the scene added for it to build the project... to do that, file>build settings>click "add open scene" (if hello cardboard is the open scene at the time). and it will appear on the list of scenes in the build.

**Comment by craigcomstock** [comment]
Yes. I found those few quirks as well and slight imprecision with the instructions. And now I have built and run the app on a phone. Works great! :) Is there somewhere I can contribute a pull request to improve the docs?
